ls1intum/Artemis
# Improve Distributed Quiz Exercise Cache Code

### Checklist

### Motivation and Context
Fix some code style issues and improve the readability, add more documentation, explain the reasoning and make the code easier understandable and maintainable.

### Description
- Removed some unnecessary explicit annotation attributes 
- Used `null` as id for the empty quiz exercise cache instead of `-1`
- Renamed some attributes, methods and classes
- **Encapsulated the quiz exercise caches and introduced a new class**
- Throw an `UnsupportedOperationException` when `clear()` is called on the empty cache.
- Added a *lot more* JavaDoc to classes, methods, fields and statements.
- Added and modified integration tests

Only the `clear()` and `null`-id changes should change functionality (that should never be used anywhere).

### Steps for Testing
- **Most important: review the code and check that it is possible to understand what it does.** Check if everything looks right.
- Execute (quiz) tests
- Test participating in Artemis Quiz Exercises, check that no submission is lost.
